### PR TITLE
Add test for encrypted ORC write [databricks]

### LINKED ISSUE #5722: ORC encryption tests for Databricks should pass
**Is your feature request related to a problem? Please describe.**
Currently we are skipping ORC encryption tests for Databricks 10.4 because they are failing.

**Describe the solution you'd like**
The tests should pass

**Describe alternatives you've considered**
Tried setting different providers e.g. `unknown`, `hadoop` and `aws`.
`unknown` results in an error saying that the provider is required. 
`hadoop` results in 
``` 
Cause: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1.0 failed 1 times, most recent failure: Lost task 0.0 in stage 1.0 (TID 1) (localhost executor driver): java.lang.IllegalArgumentException: Unknown key pii
```
`aws` results in an error saying provider url is unreachable. 

Also tried setting the provider URL to `test:///` to no avail

### PR DESCRIPTION
**Pull request by razajafri**
This PR uses the `InMemoryKeystore` to write ORC and make sure we fallback to the CPU 

* Added a test method to `SparkQueryCompareTestSuite` to capture the `DataWritingCommandExec`
* Added `OrcEncryptionSuite` to test the encrypted ORC write
* Added the `OrcEncryptionSuite` to different Spark 3.2+ profiles 

fixes #5722

### CODE REVIEW

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
build

File: tests/src/test/320+/scala/com/nvidia/spark/rapids/OrcEncryptionSuite.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      // ORC encryption is only allowed in 3.2+
+      val isValidTestForSparkVersion = SparkShimImpl.getSparkShimVersion match {
+        case SparkShimVersion(major, minor, _) => major == 3 && minor != 1
+        case DatabricksShimVersion(major, minor, _, _) => major == 3 && minor != 1
+        case ClouderaShimVersion(major, minor, _, _) => major == 3 && minor != 1
+        case _ => true
+      }
+      assume(isValidTestForSparkVersion)
+
+     val tempFile = File.createTempFile("orc-encryption-test", "")
```
Comment (tgravescs) [inline tests/src/test/320+/scala/com/nvidia/spark/rapids/OrcEncryptionSuite.scala:58]: spacing looks off here

File: tests/src/test/scala/com/nvidia/spark/rapids/SparkQueryCompareTestSuite.scala [hunk 2]
```diff
@@ -249,6 +249,57 @@ trait SparkQueryCompareTestSuite extends FunSuite with Arm {
     }
   }
 
+  def writeOnCpuAndGpuWithCapture(df: SparkSession => DataFrame,
```
Comment (tgravescs) [inline tests/src/test/scala/com/nvidia/spark/rapids/SparkQueryCompareTestSuite.scala:252]: its a bit odd this is called writeOn when this function doesn't do the write, it depends on the function passed in.  Perhaps renamed to be something more generic executeDF or something like that.

File: tests/src/test/320+/scala/com/nvidia/spark/rapids/OrcEncryptionSuite.scala [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      frame.write.options(Map("orc.key.provider" -> "memory",
```
Comment (tgravescs) [inline tests/src/test/320+/scala/com/nvidia/spark/rapids/OrcEncryptionSuite.scala:59]: so with your original change removed to fallback, this test fails?

Comment (razajafri) [reply tests/src/test/320+/scala/com/nvidia/spark/rapids/OrcEncryptionSuite.scala:59]: Yes

**Comment by razajafri** [comment]
@tgravescs PTAL

**Comment by tgravescs** [comment]
build

**Pull request merged**